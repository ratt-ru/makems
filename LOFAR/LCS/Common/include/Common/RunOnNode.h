//#  filename.h: one line description
//#
//#  Copyright (C) 2002-2004
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id: RunOnNode.h 10717 2007-12-04 13:53:32Z loose $

#ifndef LOFAR_COMMON_RUNONNODE_H
#define LOFAR_COMMON_RUNONNODE_H

// \file

//# Never #include <config.h> or #include <lofar_config.h> in a header file!

//# Includes
#include <Common/LofarLogger.h>

namespace LOFAR
{
  //# Forward Declarations

  // \addtogroup Common
  // @{
  
  class RunOnNode
  {
  public:
    RunOnNode(int procesid, int apllid=0);
    void setProcessID(int processid);
    void setApplID(int applid);

    bool mustExecute(int processid, int applid=0);
    bool mustExecuteAppl(int processid);

  private:
    static int itsProcessID;
    static int itsApplID;
  };
  
  inline void RunOnNode::setProcessID(int processid) {
    DBGASSERTSTR(processid>=0  , "processid value invallid");
    itsProcessID=processid;
  }

  inline void RunOnNode::setApplID(int applid) {
    DBGASSERTSTR(applid>=0  , "applid value invallid");
    itsApplID=applid;
  }

  inline bool RunOnNode::mustExecute(int processid, int applid) {
    bool result = ((itsProcessID == processid) && (itsApplID == applid));
    LOG_TRACE_COND_STR("mustExecute(" << processid << "," << applid << ")  :  " << result);
    return result;
  }
  inline bool RunOnNode::mustExecuteAppl(int applid) {
    bool result = (applid == itsApplID);
    LOG_TRACE_COND_STR("mustExecuteAppl(" << applid << ")  :  " << result);
    return result;
  }

#define SETNODE(processid, applid)       RunOnNode itsRON(processid,applid) 
//# todo: #define SETNODEPARM              RunOnNode itsRON(param(...),param(...))

#define RUNINPROCESS(processid, applid)  if(itsRON.mustExecute(processid,applid))
#define RIP(processid,applid)            if(itsRON.mustExecute(processid,applid))
#define RUNINAPPL(applid)                if(itsRON.mustExecuteAppl(applid))
#define RIA(applid)                      if(itsRON.mustExecuteAppl(applid))

  // @}

} // namespace LOFAR

#endif
