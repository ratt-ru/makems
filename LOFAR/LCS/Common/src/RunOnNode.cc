//#  filename.cc: one line description
//#
//#  Copyright (C) 2002-2004
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id: RunOnNode.cc 4847 2005-04-13 14:33:43Z loose $

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>

#include <Common/RunOnNode.h>

namespace LOFAR
{

int  RunOnNode::itsProcessID=0;
int  RunOnNode::itsApplID=0;
  
  RunOnNode::RunOnNode(int processid, int applid) {
    itsProcessID = processid;
    itsApplID    = applid;
  }


} // namespace LOFAR
