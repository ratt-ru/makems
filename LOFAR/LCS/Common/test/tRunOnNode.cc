//#  testRunOnNode.cc: Program to test the interface to the LofarLogger.
//#
//#  Copyright (C) 2004
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id: tRunOnNode.cc 7376 2006-01-03 14:50:34Z loose $

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>


#include <Common/RunOnNode.h>

using namespace std;
using namespace LOFAR;

int main (int, const char**) {

  cout << "Program started" << endl;
  
  cout << "Initialise RunOnNode; tell this process to be (0,0)" << endl;
  SETNODE(0,0);
  
  cout << "Check for conditional execution" << endl;
  RUNINPROCESS(0,0)  cout << "Executing code for (0,0)" << endl;
  RUNINPROCESS(3,7)  cout << "ERROR: Executing code for (3,7)" << endl;

// todo:test reset of node (not implemented yet)
// 	cout << "Now re-init to (3,7)" << endl;
// 	SETNODE(3,7);
// 	cout << "Check for conditional execution" << endl;
// 	RUNINPROCESS(0,0)  cout << "ERROR: Executing code for (0,0)" << endl;
// 	RUNINPROCESS(3,7)  cout << "Executing code for (3,7)" << endl;

	
}

