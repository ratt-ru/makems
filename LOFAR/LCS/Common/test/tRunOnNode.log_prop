# Property file to be use with the demo program testLogger.

# Configure the rootLogger
log4cplus.rootLogger=DEBUG, STDOUT
# Define the STDOUT appender
log4cplus.appender.STDOUT=log4cplus::ConsoleAppender
log4cplus.appender.STDOUT.layout=log4cplus::PatternLayout
log4cplus.appender.STDOUT.layout.ConversionPattern=%-5p [%x]%c{3} - %m%n

# Define foo at level TRACE4
#log4cplus.logger.TRC.foo=TRACE4

