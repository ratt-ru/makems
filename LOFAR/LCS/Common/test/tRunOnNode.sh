#!/bin/sh
./runctest.sh tRunOnNode 2>&1 > tRunOnNode.log
