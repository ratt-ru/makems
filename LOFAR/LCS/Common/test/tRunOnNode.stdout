Program started
Initialise RunOnNode; tell this process to be (0,0)
Check for conditional execution
Executing code for (0,0)
